packages/setup-ocaml/src/opam.ts

```diff
@@ -120,6 +120,22 @@ export async function setupOpam() {
   await initializeOpam();
 }
 
+export function setupOpamEnv() {
+  const githubWorkspace = process.env.GITHUB_WORKSPACE ?? process.cwd();
+  const OPAM_SWITCH_PREFIX = path.join(githubWorkspace, "_opam");
+  const CAML_LD_LIBRARY_PATH = [
+    path.join(OPAM_SWITCH_PREFIX, "lib", "stublibs"),
+    path.join(OPAM_SWITCH_PREFIX, "lib", "ocaml", "stublibs"),
+    path.join(OPAM_SWITCH_PREFIX, "lib", "ocaml"),
+  ].join(path.delimiter);
+  const OCAML_TOPLEVEL_PATH = path.join(OPAM_SWITCH_PREFIX, "lib", "toplevel");
+  const OPAM_SWITCH_BIN = path.join(OPAM_SWITCH_PREFIX, "bin");
+  core.exportVariable("OPAM_SWITCH_PREFIX", OPAM_SWITCH_PREFIX);
+  core.exportVariable("CAML_LD_LIBRARY_PATH", CAML_LD_LIBRARY_PATH);
+  core.exportVariable("OCAML_TOPLEVEL_PATH", OCAML_TOPLEVEL_PATH);
+  core.addPath(OPAM_SWITCH_BIN);
+}
+
```

Comment:
This doesn't work, I'm afraid - you either need to run `opam env` and parse its output, not guess the values.

FWIW, I think it's better for CI to have to be explicit about using `opam exec`

Reply:
Parsing the output sounds a bit overkill. The commit was dropped.

Reply:
@dra27 For the record, could you explain why this approach doesn't work?